### PR TITLE
updates from issue 61

### LINKED ISSUE #61: use `teal.devel::plot_with_settings` for static plots
For all static graph type of the outputs please consider using `teal.widgets::plot_with_settings` functionality (you won't find this function - there are two with `_srv` and `_ui` suffixes). It gives nice features such as download, resize and modal preview. 
image

The only requirement is to pass on a plot in form of reactive expression. For details / example please see the docs, as well as https://github.com/insightsengineering/teal.modules.hermes/blob/main/R/forestplot.R where it is already used.

To do:
- [x] Barplot
- [x] Boxplot
- [x] Kaplan-Meier Plot
- [ ] PCA plots
- [x] Quality module plots
- [x] Scatterplot
- [ ] Volcanoplot

### PR DESCRIPTION
**Pull request by sob2021**
closes #61

### CODE REVIEW

File: R/pca.R [hunk 1]
```diff
@@ -410,6 +421,7 @@ srv_g_pca <- function(id,
           card$append_text(final_encodings, style = "verbatim")
           card$append_text("Plot", "header3")
           card$append_plot(plot_cor())
+          # card$append_plot(plot_cor(), dim = pws_cor$dim())
```
Comment (danielinteractive) [inline R/pca.R:424]: need to remove commented code - but maybe we need the dim arg here?

File: R/volcanoplot.R [hunk 2]
```diff
@@ -230,10 +241,11 @@ srv_g_volcanoplot <- function(id,
 
         card$append_text(final_encodings, style = "verbatim")
         card$append_text("Plot", "header3")
-        card$append_plot(plot_r())
+        card$append_plot(plot_r(), dim = pws_p$dim())
         if (isTRUE(input$show_top_gene)) {
           card$append_text("Table", "header3")
           card$append_table(show_top_gene_diffexpr())
+          # card$append_table(show_top_gene_diffexpr(), dim = pws_t$dim())
```
Comment (danielinteractive) [inline R/volcanoplot.R:248]: same here

**Review by danielinteractive — COMMENTED** [review]
thanks @sob2021 , do we foresee that we need updates in the shiny tests? @BFalquet

File: R/pca.R [hunk 3]
```diff
@@ -127,7 +127,7 @@ ui_g_pca <- function(id,
               width = 12,
               div(
                 class = "my-5",
-                plotOutput(ns("plot_pca"))
+                teal.widgets::plot_with_settings_ui(ns("plot_pca"))
               ),
               DT::DTOutput(ns("table_pca"))
```
Comment (danielinteractive) [inline R/pca.R:132]: @Polkas one question, how does the `table_with_settings()` work, do we have an example from another teal module? we tried to use here but gave an error, and I just tried to find another teal module but did not find any for some reason, thanks

File: R/pca.R [hunk 4]
```diff
@@ -139,6 +139,7 @@ ui_g_pca <- function(id,
               div(
                 class = "my-5",
                 plotOutput(ns("plot_cor"))
+                # teal.widgets::plot_with_settings_ui(ns("plot_cor"))
```
[the remark this answers is not in the document]
Comment (Polkas) [reply R/pca.R]: I simply remove `plotOutput(ns("plot_cor"))` and uncomment the plot with settings calls and everything works fine for me now.

[the remark this answers is not in the document]
Comment (Polkas) [reply R/pca.R]: I commit the update.

File: R/pca.R (hunk 3 above)
Comment (Polkas) [reply R/pca.R:132]: I see the documentation for table_with_settings is very poor. It looks like only rtables object could be used but you want to pass a regular data.frame to `teal.widgets::table_with_settings_srv`.

Almost all tmc modules for tables (tm_t_*) are returning rtables and printing it with table with settings.
e.g. https://github.com/insightsengineering/teal.modules.clinical/blob/main/R/tm_t_events.R

Comment (Polkas) [reply R/pca.R:132]: a new issue https://github.com/insightsengineering/teal.widgets/issues/101

File: R/volcanoplot.R (hunk 2 above)
Comment (Polkas) [reply R/volcanoplot.R:248]: append_table method does not have the dim argument so should be removed.

Comment (Polkas) [reply R/volcanoplot.R]: I remove the comment but be very careful as the table easily could be ver long and take to much space in the report. May be the head + tail of the table is enough?

File: R/volcanoplot.R [hunk 5]
```diff
@@ -196,6 +202,11 @@ srv_g_volcanoplot <- function(id,
       )
     })
 
+    # pws_t <- teal.widgets::table_with_settings_srv(
+    #   id = "table",
+    #   table_r = show_top_gene_diffexpr
+    # )
+
```
Comment (Polkas) [inline R/volcanoplot.R:209]:
```suggestion
```

File: R/volcanoplot.R [hunk 6]
```diff
@@ -69,8 +69,9 @@ ui_g_volcanoplot <- function(id,
 
   teal.widgets::standard_layout(
     output = div(
-      plotOutput(ns("plot")),
+      teal.widgets::plot_with_settings_ui(ns("plot")),
       DT::DTOutput(ns("table"))
+      # teal.widgets::table_with_settings_ui(ns("table"))
```
Comment (Polkas) [inline R/volcanoplot.R:74]:
```suggestion
```

File: R/volcanoplot.R [hunk 7]
```diff
@@ -230,10 +241,11 @@ srv_g_volcanoplot <- function(id,
 
         card$append_text(final_encodings, style = "verbatim")
         card$append_text("Plot", "header3")
-        card$append_plot(plot_r())
+        card$append_plot(plot_r(), dim = pws_p$dim())
         if (isTRUE(input$show_top_gene)) {
           card$append_text("Table", "header3")
           card$append_table(show_top_gene_diffexpr())
+          card$append_table(show_top_gene_diffexpr())
```
Comment (Polkas) [inline R/volcanoplot.R:248]:
```suggestion
```

File: R/volcanoplot.R [hunk 8]
```diff
@@ -230,7 +241,7 @@ srv_g_volcanoplot <- function(id,
 
         card$append_text(final_encodings, style = "verbatim")
         card$append_text("Plot", "header3")
-        card$append_plot(plot_r())
+        card$append_plot(plot_r(), dim = pws_p$dim())
         if (isTRUE(input$show_top_gene)) {
           card$append_text("Table", "header3")
           card$append_table(show_top_gene_diffexpr())
```
Comment (Polkas) [inline R/volcanoplot.R:247]: reporter supports data.frame and rtables at the same time.

File: R/pca.R (hunk 3 above)
Comment (danielinteractive) [reply R/pca.R:132]: ok thanks @Polkas then we will keep the code as is here - the `DT` is nice here for quick exploration of the data frame, so we don't want to revert that to `rtables`

File: R/volcanoplot.R (hunk 8 above)
Comment (danielinteractive) [reply R/volcanoplot.R:247]: ok that is great, good to know!

**Review by danielinteractive — APPROVED** [review]

**Pull request merged**